### PR TITLE
User/mtian/create cohort no login

### PR DESCRIPTION
**Pull request by mitianisb**
This pull request related to the tickets #2758, #2757, #2744, #2745 and implemented following features.
1. Added "Data Explorer (New)" to the webpage header, and to the landing page. This brings user to the cohort creation page and it can be used without logging in.
2. The save cohort button will show as "Log In To Save New Cohort" if the user has not logged in, and as "Save As New Cohort" if the user has logged in.
3. Disabled the "USER DATA" tab if the user has not logged in.
4. If user choose "Log In To Save New Cohort", it will redirect to the same cohort creation page after log in instead of redirect to dashboard page.
5. After redirected back, the previously selected filters before log in will be restored, so they do not need to redo the whole process again.

### CODE REVIEW

File: static/js/cohort_details.js [hunk 1]
```diff
@@ -1093,47 +1105,157 @@ require([
...
+                checkboxId = checkboxId.toUpperCase();
+
+                // Get checkbox by id, jQuery does not work for some special characters so using DOM here
+                var checkbox = document.getElementById(checkboxId);
+
+                if (checkbox != null) {
+                    // Set checked and trigger change to update other related data
+                    checkbox.checked = true;
+                    var event = new Event('change');
+                    checkbox.dispatchEvent(event);
```
Comment (s-paquette) [inline static/js/cohort_details.js]: The issue here is this triggers the update callback once for each filter value - a serious issue if a user clicked multiple filters. A better way to do this is to track which filter you're on in the set, and on the last one, pass a special value to trigger the update only then. Doing this requires a couple small changes:

- Add a parameter to the `filter_change_callback` event handler, eg. withoutDisplayUpdates. No default value needed.
- At the end of `filter_change_callback`, add `!withoutDisplayUpdates &&` in front of `update_displays()`
- In your calls here, you do something like: `$(checkboxId).trigger('change', [Boolean(i !== (ANONYMOUS_FILTERS.length-1))])` (Example assumes the checkbox was already set to checked and the name constructed as a selector ala `#2-VITAL_STATUS-ALIVE`)

File: static/js/cohort_details.js [hunk 2]
```diff
@@ -1093,47 +1105,157 @@ require([
...
+                var featureId = aFilter.feature.id.toString();
+                var valueId = aFilter.value.id.toString();
+
+                var checkboxId = programId + "-" + featureId + "-" + valueId;
+                checkboxId = checkboxId.replace(/ /g, "_");
+                checkboxId = checkboxId.toUpperCase();
+
+                // Get checkbox by id, jQuery does not work for some special characters so using DOM here
+                var checkbox = document.getElementById(checkboxId);
+
```
[the remark this answers is not in the document]
Comment (mitianisb) [reply static/js/cohort_details.js]: Thank you for the careful review! These are some of special characters that make the jQuery method not work when I tried:
1. "." - under TARGET, WBC AT DIAGNOSIS. I tried to use \\ to escape but it does not work I am not sure why.
2. "&", ":", "/" - in TCGA, HISTOLOGICAL TYPE, Head & Neck.
I think if we can use jQuery it is better than what I am using now.

[the remark this answers is not in the document]
Comment (mitianisb) [reply static/js/cohort_details.js]: Hi Suzanne, I found some code on the internet and it seems working:
                checkboxId = checkboxId.replace(/([$%&()*+,./:;<=>?@\[\\\]^\{|}~])/g, '\\$1');
I'll update the code. Please let me know if this is correct.

File: static/js/cohort_details.js (hunk 1 above)
Comment (mitianisb) [reply static/js/cohort_details.js]: Good Point.  I have changed the code. Thanks!

File: static/js/cohort_details.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (s-paquette) [reply static/js/cohort_details.js]: Ah ha, yeah that would be an issue. And that solution works with one small change--you want to use two slashes, not just one. so: `replace(/([$%&()*+,./:;<=>?@[\]^{|}~])/g, '\\$1');`

**Comment by mitianisb** [comment]
> Code looks good to me, Mi!
> The only thing I'd point out is I think **is_logged_in** can be replaced to **user.is_authenticated**

Thanks! I have addressed your comment.

**Pull request merged**